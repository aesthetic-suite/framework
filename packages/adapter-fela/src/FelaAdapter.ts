import {
  Adapter,
  ClassName,
  Ruleset,
  Sheet,
  StyleName,
  GLOBAL_STYLE_NAME,
  FontFace,
} from 'aesthetic';
import { getStyleElements, purgeStyles, toArray } from 'aesthetic-utils';
import { combineRules, IRenderer } from 'fela';
import { render } from 'fela-dom';
import { NativeBlock, ParsedBlock } from './types';

export default class FelaAdapter extends Adapter<NativeBlock, ParsedBlock> {
  fela: IRenderer;

  keyframes: { [animationName: string]: ClassName } = {};

  constructor(fela: IRenderer) {
    super();

    this.fela = fela;

    this.syntax
      .on('attribute', this.handleNested)
      .on('css', this.handleCss)
      .on('fallback', this.handleFallback)
      .on('font-face', this.handleFontFace)
      .on('global', this.handleGlobal)
      .on('keyframe', this.handleKeyframe)
      .on('media', this.handleMedia)
      .on('property', this.handleProperty)
      .on('pseudo', this.handleNested)
      .on('selector', this.handleNested)
      .on('support', this.handleSupport);
  }

  flushStyles() {
    render(this.fela);
  }

  transformToClassName(styles: ParsedBlock[]): ClassName {
    return this.fela.renderRule(combineRules(...styles.map(style => () => style)), {});
  }

  purgeStyles(styleName?: StyleName) {
    purgeStyles(getStyleElements('data-fela-type'), styleName === GLOBAL_STYLE_NAME);
  }

  // http://fela.js.org/docs/api/fela/Renderer.html
  private handleCss = (css: string) => {
    this.fela.renderStatic(css);
  };

  // https://github.com/rofrischmann/fela/tree/master/packages/fela-plugin-fallback-value
  private handleFallback = (
    ruleset: Ruleset<NativeBlock>,
    name: keyof NativeBlock,
    value: unknown[],
  ) => {
    ruleset.addProperty(name, [...value, ...toArray(ruleset.properties[name])]);
  };

  // http://fela.js.org/docs/basics/Fonts.html
  // http://fela.js.org/docs/api/fela/Renderer.html
  private handleFontFace = (
    sheet: Sheet<NativeBlock>,
    fontFaces: Ruleset<NativeBlock>[],
    fontFamily: string,
    srcPaths: string[][],
  ) => {
    fontFaces.forEach((face, i) => {
      const { local, ...style } = face.toObject() as FontFace;

      this.fela.renderFont(fontFamily, srcPaths[i], {
        ...style,
        localAlias: local,
      });
    });
  };

  // http://fela.js.org/docs/advanced/StaticStyle.html
  // http://fela.js.org/docs/api/fela/Renderer.html#renderstaticstyle-selector
  private handleGlobal = (
    sheet: Sheet<NativeBlock>,
    selector: string,
    ruleset: Ruleset<NativeBlock>,
  ) => {
    this.fela.renderStatic(ruleset.toObject(), selector);
  };

  // http://fela.js.org/docs/basics/Keyframes.html
  // http://fela.js.org/docs/basics/Renderer.html#renderkeyframe
  private handleKeyframe = (
    sheet: Sheet<NativeBlock>,
    keyframe: Ruleset<NativeBlock>,
    animationName: string,
  ) => {
    this.keyframes[animationName] = this.fela.renderKeyframe(
      // @ts-ignore Annoying to type
      () => keyframe.toObject(),
      {},
    );
  };

  // http://fela.js.org/docs/basics/Rules.html#3-media-queries
  private handleMedia = (
    ruleset: Ruleset<NativeBlock>,
    query: string,
    value: Ruleset<NativeBlock>,
  ) => {
    ruleset.addNested(`@media ${query}`, value);
  };

  // http://fela.js.org/docs/basics/Rules.html#style-object
  private handleNested = (
    ruleset: Ruleset<NativeBlock>,
    selector: string,
    value: Ruleset<NativeBlock>,
  ) => {
    ruleset.addNested(selector, value);
  };

  // http://fela.js.org/docs/basics/Rules.html
  private handleProperty = (
    ruleset: Ruleset<NativeBlock>,
    name: keyof NativeBlock,
    value: unknown,
  ) => {
    if (name === 'animationName') {
      ruleset.addProperty(
        name,
        this.syntax.injectKeyframes(String(value), this.keyframes).join(', '),
      );
    } else {
      ruleset.addProperty(name, value);
    }
  };

  // http://fela.js.org/docs/basics/Rules.html
  private handleSupport = (
    ruleset: Ruleset<NativeBlock>,
    query: string,
    value: Ruleset<NativeBlock>,
  ) => {
    ruleset.addNested(`@supports ${query}`, value);
  };
}
