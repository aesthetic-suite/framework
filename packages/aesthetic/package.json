{
  "name": "aesthetic",
  "version": "2.6.0",
  "description": "Aesthetic is a powerful React library for styling components through the use of adapters.",
  "keywords": [
    "aesthetic",
    "react",
    "native",
    "style",
    "components",
    "hoc",
    "css",
    "css-in-js",
    "css-modules",
    "aphrodite",
    "glamor",
    "fela",
    "jss"
  ],
  "repository": "https://github.com/milesj/aesthetic/tree/master/packages/aesthetic",
  "license": "MIT",
  "main": "./lib/index.js",
  "types": "./lib/index.d.ts",
  "publishConfig": {
    "access": "public"
  },
  "dependencies": {
    "hoist-non-react-statics": "^2.5.5",
    "lodash": "^4.17.10",
    "prop-types": "^15.6.2"
  },
  "peerDependencies": {
    "react": "^15.0.0 || ^16.0.0"
  },
  "devDependencies": {
    "react": "^16.4.1"
  }
}
