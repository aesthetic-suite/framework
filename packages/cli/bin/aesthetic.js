#!/usr/bin/env node

try {
  // Attempt to load it incase some dependencies require it
  // eslint-disable-next-line
  require('regenerator-runtime');
} catch {
  // Ignore if missing
}

require('../lib');
