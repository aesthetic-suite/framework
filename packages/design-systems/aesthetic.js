module.exports = require('./lib/aesthetic/themes/default.js');
