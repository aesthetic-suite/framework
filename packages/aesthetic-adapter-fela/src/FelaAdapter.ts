/**
 * @copyright   2017, Miles Johnson
 * @license     https://opensource.org/licenses/MIT
 */

import { Adapter, ClassName, UnifiedSyntax, FontFace, Keyframes, injectKeyframes } from 'aesthetic';
import { createRenderer, combineRules, IRenderer } from 'fela';
import { render } from 'fela-dom';
import { StyleSheet, Declaration } from './types';

export default class FelaAdapter extends Adapter<StyleSheet, Declaration> {
  fela: IRenderer;

  keyframesCache: { [animationName: string]: string } = {};

  constructor(fela?: IRenderer) {
    super();

    this.fela = fela || createRenderer();

    render(this.fela);
  }

  transform(...styles: Declaration[]): ClassName {
    return this.fela.renderRule(
      combineRules(...styles.map(style => (typeof style === 'function' ? style : () => style))),
      {},
    );
  }

  unify(syntax: UnifiedSyntax<StyleSheet, Declaration>) {
    syntax
      .on('property', this.handleProperty)
      .on('@charset', syntax.createUnsupportedHandler('@charset'))
      .on('@font-face', this.handleFontFace)
      .on('@global', this.handleGlobal)
      .on('@import', syntax.createUnsupportedHandler('@import'))
      .on('@keyframes', this.handleKeyframe)
      .on('@namespace', syntax.createUnsupportedHandler('@namespace'))
      .on('@page', syntax.createUnsupportedHandler('@page'))
      .on('@viewport', syntax.createUnsupportedHandler('@viewport'));
  }

  // http://fela.js.org/docs/basics/Fonts.html
  // http://fela.js.org/docs/basics/Renderer.html#renderfont
  handleFontFace = (
    styleSheet: StyleSheet,
    fontFaces: FontFace[],
    fontFamily: string,
    srcPaths: string[][],
  ) => {
    fontFaces.map((face, i) => this.fela.renderFont(fontFamily, srcPaths[i], face));
  };

  // http://fela.js.org/docs/advanced/StaticStyle.html
  // http://fela.js.org/docs/api/fela/Renderer.html#renderstaticstyle-selector
  handleGlobal = (styleSheet: StyleSheet, declaration: Declaration, selector: string) => {
    this.fela.renderStatic(declaration, selector);
  };

  // http://fela.js.org/docs/basics/Keyframes.html
  // http://fela.js.org/docs/basics/Renderer.html#renderkeyframe
  handleKeyframe = (styleSheet: StyleSheet, keyframes: Keyframes, animationName: string) => {
    this.keyframesCache[animationName] = this.fela.renderKeyframe(() => keyframes);
  };

  handleProperty = (declaration: Declaration, value: any, property: string) => {
    if (property === 'animationName') {
      declaration[property] = injectKeyframes(value, this.keyframesCache).join(', ');
    } else {
      declaration[property] = style;
    }
  };
}
