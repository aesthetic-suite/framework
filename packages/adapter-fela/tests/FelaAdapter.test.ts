/* eslint-disable jest/expect-expect */

import { createRenderer } from 'fela';
import webPreset from 'fela-preset-web';
import { Aesthetic, GLOBAL_STYLE_NAME } from 'aesthetic';
import {
  setupAesthetic,
  cleanupStyleElements,
  getFlushedStyles,
  renderAndExpect,
  DIRECTIONS,
  FONT_ROBOTO_FLAT_SRC,
  FONT_CIRCULAR_MULTIPLE_FLAT_SRC,
  KEYFRAME_FADE,
  SYNTAX_UNIFIED_LOCAL_FULL,
  SYNTAX_ATTRIBUTE,
  SYNTAX_DESCENDANT,
  SYNTAX_FALLBACKS,
  SYNTAX_FONT_FACE,
  SYNTAX_FONT_FACE_MIXED,
  SYNTAX_FONT_FACE_MULTIPLE,
  SYNTAX_GLOBAL,
  SYNTAX_KEYFRAMES,
  SYNTAX_MEDIA_QUERY,
  SYNTAX_MULTI_SELECTOR,
  SYNTAX_PROPERTIES,
  SYNTAX_PSEUDO,
  SYNTAX_SUPPORTS,
  FONT_ROBOTO,
  FONT_CIRCULAR_MULTIPLE,
  SYNTAX_MEDIA_QUERY_NESTED,
  SYNTAX_KEYFRAMES_INLINE,
  SYNTAX_FONT_FACES_INLINE,
  SYNTAX_RAW_CSS,
} from 'aesthetic/lib/testing';
import FelaAdapter from '../src/FelaAdapter';

describe('FelaAdapter', () => {
  let instance: FelaAdapter;

  beforeEach(() => {
    instance = new FelaAdapter(
      createRenderer({
        plugins: [...webPreset],
      }),
    );

    setupAesthetic(new Aesthetic(), instance);
  });

  afterEach(() => {
    cleanupStyleElements();
  });

  DIRECTIONS.forEach(dir => {
    describe(`${dir.toUpperCase()}`, () => {
      it('converts and transforms inline styles', () => {
        expect(instance.transformStyles([{ margin: 0 }, { padding: 2 }], { dir })).toBe('a b');
      });

      describe('global sheet', () => {
        it('flushes and purges global styles from the DOM', () => {
          renderAndExpect(instance, SYNTAX_GLOBAL, {}, { dir, global: true });

          instance.purgeStyles(GLOBAL_STYLE_NAME);

          expect(getFlushedStyles()).toMatchSnapshot();
        });

        it('handles @font-face', () => {
          const spy = jest.spyOn(instance.fela, 'renderFont');

          instance.syntax.convertGlobalSheet(SYNTAX_FONT_FACE, { dir });

          expect(spy).toHaveBeenCalledWith('Roboto', FONT_ROBOTO.srcPaths, FONT_ROBOTO_FLAT_SRC);
          expect(spy).toHaveBeenCalledTimes(1);
        });

        it('handles mixed @font-face', () => {
          const spy = jest.spyOn(instance.fela, 'renderFont');

          instance.syntax.convertGlobalSheet(SYNTAX_FONT_FACE_MIXED, { dir });

          expect(spy).toHaveBeenCalledWith('Roboto', FONT_ROBOTO.srcPaths, FONT_ROBOTO_FLAT_SRC);
          expect(spy).toHaveBeenCalledWith(
            'Circular',
            FONT_CIRCULAR_MULTIPLE[0].srcPaths,
            FONT_CIRCULAR_MULTIPLE_FLAT_SRC[0],
          );
          expect(spy).toHaveBeenCalledTimes(5);
        });

        it('handles multiple @font-face', () => {
          const spy = jest.spyOn(instance.fela, 'renderFont');

          instance.syntax.convertGlobalSheet(SYNTAX_FONT_FACE_MULTIPLE, { dir });

          expect(spy).toHaveBeenCalledWith(
            'Circular',
            FONT_CIRCULAR_MULTIPLE[0].srcPaths,
            FONT_CIRCULAR_MULTIPLE_FLAT_SRC[0],
          );
          expect(spy).toHaveBeenCalledTimes(4);
        });

        it('handles @keyframes', () => {
          const spy = jest.spyOn(instance.fela, 'renderKeyframe');

          instance.syntax.convertGlobalSheet(SYNTAX_KEYFRAMES, { dir });

          expect(spy).toHaveBeenCalledWith(expect.anything(), {});
        });
      });

      describe('style sheet', () => {
        it('flushes and purges all styles from the DOM', () => {
          const styles = { test: { display: 'block' } };

          renderAndExpect(instance, styles, styles, { dir });

          instance.purgeStyles();

          expect(getFlushedStyles()).toMatchSnapshot();
        });

        it('converts unified syntax to native syntax and transforms to a class name', () => {
          instance.fela.renderFont('Roboto', FONT_ROBOTO.srcPaths, FONT_ROBOTO);

          // @ts-ignore
          instance.keyframes.fade = instance.fela.renderKeyframe(() => KEYFRAME_FADE, {});

          renderAndExpect(
            instance,
            SYNTAX_UNIFIED_LOCAL_FULL,
            {
              button: {
                margin: 0,
                padding: '6px 12px',
                border: '1px solid #2e6da4',
                borderRadius: 4,
                display: 'inline-block',
                cursor: 'pointer',
                fontFamily: 'Roboto',
                fontWeight: 'normal',
                lineHeight: 'normal',
                whiteSpace: 'nowrap',
                textDecoration: 'none',
                textAlign: 'left',
                backgroundColor: '#337ab7',
                verticalAlign: 'middle',
                color: 'rgba(0, 0, 0, 0)',
                animationName: 'k1',
                animationDuration: '.3s',
                ':hover': {
                  backgroundColor: '#286090',
                  borderColor: '#204d74',
                },
                '::before': {
                  content: '"â˜…"',
                  display: 'inline-block',
                  verticalAlign: 'middle',
                  marginRight: 5,
                },
                '@media (max-width: 600px)': {
                  padding: '4px 8px',
                },
              },
            },
            { dir },
          );
        });

        it('handles properties', () => {
          renderAndExpect(instance, SYNTAX_PROPERTIES, SYNTAX_PROPERTIES, { dir });
        });

        it('handles attribute selectors', () => {
          renderAndExpect(instance, SYNTAX_ATTRIBUTE, SYNTAX_ATTRIBUTE, { dir });
        });

        it('handles descendant selectors', () => {
          renderAndExpect(
            instance,
            SYNTAX_DESCENDANT,
            {
              list: {
                margin: 0,
                padding: 0,
                '> li': {
                  listStyle: 'bullet',
                },
              },
            },
            { dir },
          );
        });

        it('handles pseudo selectors', () => {
          renderAndExpect(instance, SYNTAX_PSEUDO, SYNTAX_PSEUDO, { dir });
        });

        it('handles multiple selectors (comma separated)', () => {
          renderAndExpect(
            instance,
            SYNTAX_MULTI_SELECTOR,
            {
              multi: {
                cursor: 'pointer',
                ':disabled': { cursor: 'default' },
                '[disabled]': { cursor: 'default' },
                '> span': { cursor: 'default' },
              },
            },
            { dir },
          );
        });

        it('handles inline @keyframes', () => {
          renderAndExpect(
            instance,
            SYNTAX_KEYFRAMES_INLINE,
            {
              single: {
                animationName: 'k1',
              },
              multiple: {
                animationName: 'k1, unknown, k2',
              },
            },
            { dir },
          );
        });

        it('handles inline @font-face', () => {
          renderAndExpect(
            instance,
            SYNTAX_FONT_FACES_INLINE,
            {
              single: {
                fontFamily: 'Roboto',
              },
              multiple: {
                fontFamily: 'Circular, OtherFont, Roboto',
              },
            },
            { dir },
          );
        });

        it('handles @fallbacks', () => {
          renderAndExpect(
            instance,
            SYNTAX_FALLBACKS,
            {
              fallback: {
                background: ['red', 'linear-gradient(...)'],
                display: ['block', 'inline-block', 'flex'],
                color: ['blue'],
              },
            },
            { dir },
          );
        });

        it('handles @media', () => {
          renderAndExpect(
            instance,
            SYNTAX_MEDIA_QUERY,
            {
              media: {
                color: 'red',
                paddingLeft: 10,
                '@media (max-width: 1000px)': {
                  color: 'green',
                  paddingLeft: 20,
                },
                '@media (min-width: 300px)': {
                  color: 'blue',
                  paddingLeft: 15,
                },
              },
            },
            { dir },
          );
        });

        it('handles nested @media', () => {
          renderAndExpect(
            instance,
            SYNTAX_MEDIA_QUERY_NESTED,
            {
              media: {
                color: 'red',
                '@media (min-width: 300px)': {
                  color: 'blue',
                  '@media (max-width: 1000px)': {
                    color: 'green',
                  },
                },
              },
            },
            { dir },
          );
        });

        it('handles @supports', () => {
          renderAndExpect(
            instance,
            SYNTAX_SUPPORTS,
            {
              sup: {
                display: 'block',
                '@supports (display: flex)': {
                  display: 'flex',
                },
                '@supports not (display: flex)': {
                  float: 'left',
                },
              },
            },
            { dir },
          );
        });

        it('handles raw CSS', () => {
          renderAndExpect(instance, SYNTAX_RAW_CSS, {}, { dir });
        });
      });
    });
  });
});
