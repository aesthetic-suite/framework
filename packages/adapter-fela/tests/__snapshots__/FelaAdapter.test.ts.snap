// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FelaAdapter LTR global sheet flushes and purges global styles from the DOM 1`] = `
"body {margin: 0;}
html {height: 100%;}
a {color: red;}
ul {margin: 0;}"
`;

exports[`FelaAdapter LTR global sheet flushes and purges global styles from the DOM 2`] = `""`;

exports[`FelaAdapter LTR global sheet flushes and purges global styles from the DOM 3`] = `""`;

exports[`FelaAdapter LTR style sheet converts unified syntax to native syntax and transforms to a class name 1`] = `
"@-webkit-keyframes k1 { 
  from {opacity: 0;} 
  to {opacity: 1;} 
}
@-moz-keyframes k1 { 
  from {opacity: 0;} 
  to {opacity: 1;} 
}
@keyframes k1 { 
  from {opacity: 0;} 
  to {opacity: 1;} 
}
@font-face {font-family: \\"Roboto\\"; font-style: normal; font-weight: normal; src: url('fonts/Roboto.woff2') format('woff2'),url('fonts/Roboto.ttf') format('truetype');}
.a {margin: 0;}
.b {padding: 6px 12px;}
.c {border: 1px solid #2e6da4;}
.d {border-radius: 4px;}
.e {display: inline-block;}
.f {cursor: pointer;}
.g {font-family: Roboto;}
.h {font-weight: normal;}
.i {line-height: normal;}
.j {white-space: nowrap;}
.k {text-decoration: none;}
.l {text-align: left;}
.m {background-color: #337ab7;}
.n {vertical-align: middle;}
.o {color: rgba(0, 0, 0, 0);}
.p {animation-name: k1;}
.q {animation-duration: .3s;}
.t::before {content: \\"★\\";}
.u::before {display: inline-block;}
.v::before {vertical-align: middle;}
.w::before {margin-right: 5px;}
.r:hover {background-color: #286090;}
.s:hover {border-color: #204d74;}
@media (max-width: 600px) { .x {padding: 4px 8px;} }"
`;

exports[`FelaAdapter LTR style sheet converts unified syntax to native syntax and transforms to a class name 2`] = `"a b c d e f g h i j k l m n o p q r s t u v w x"`;

exports[`FelaAdapter LTR style sheet flushes and purges all styles from the DOM 1`] = `".a {display: block;}"`;

exports[`FelaAdapter LTR style sheet flushes and purges all styles from the DOM 2`] = `"a"`;

exports[`FelaAdapter LTR style sheet flushes and purges all styles from the DOM 3`] = `""`;

exports[`FelaAdapter LTR style sheet handles @fallbacks 1`] = `
".a {background: linear-gradient(...);}
.b {display: flex;}
.c {color: blue;}"
`;

exports[`FelaAdapter LTR style sheet handles @fallbacks 2`] = `"a b c"`;

exports[`FelaAdapter LTR style sheet handles @media 1`] = `
".a {color: red;}
.b {padding-left: 10px;}
@media (min-width: 300px) { .c {color: blue;}
.d {padding-left: 15px;} }
@media (max-width: 1000px) { .e {color: green;}
.f {padding-left: 20px;} }"
`;

exports[`FelaAdapter LTR style sheet handles @media 2`] = `"a b c d e f"`;

exports[`FelaAdapter LTR style sheet handles @supports 1`] = `
".a {display: block;}
@supports not (display: flex) {.c {float: left;}}
@supports (display: flex) {.b {display: flex;}}"
`;

exports[`FelaAdapter LTR style sheet handles @supports 2`] = `"a b c"`;

exports[`FelaAdapter LTR style sheet handles attribute selectors 1`] = `
".a {display: block;}
.b[disabled] {opacity: 0.5;}"
`;

exports[`FelaAdapter LTR style sheet handles attribute selectors 2`] = `"a b"`;

exports[`FelaAdapter LTR style sheet handles descendant selectors 1`] = `
".a {margin: 0;}
.b {padding: 0;}
.c> li {list-style: bullet;}"
`;

exports[`FelaAdapter LTR style sheet handles descendant selectors 2`] = `"a b c"`;

exports[`FelaAdapter LTR style sheet handles inline @font-face 1`] = `
"@font-face {font-family: \\"Roboto\\"; font-style: normal; font-weight: normal; src: url('fonts/Roboto.woff2') format('woff2'),url('fonts/Roboto.ttf') format('truetype');}
@font-face {font-family: \\"Circular\\"; font-style: normal; font-weight: normal; src: url('fonts/Circular.woff2') format('woff2');}
@font-face {font-family: \\"Circular\\"; font-style: italic; font-weight: normal; src: url('fonts/Circular-Italic.woff2') format('woff2');}
@font-face {font-family: \\"Circular\\"; font-style: normal; font-weight: 300; src: url('fonts/Circular-Light.woff2') format('woff2');}
@font-face {font-family: \\"Circular\\"; font-style: normal; font-weight: 700; src: url('fonts/Circular-Bold.woff2') format('woff2');}
.a {font-family: Circular, OtherFont, Roboto;}"
`;

exports[`FelaAdapter LTR style sheet handles inline @font-face 2`] = `"a"`;

exports[`FelaAdapter LTR style sheet handles inline @keyframes 1`] = `
"@-webkit-keyframes k1 { 
  0% {left: 0%;} 
  50% {left: 50%;} 
  100% {left: 100%;} 
}
@-moz-keyframes k1 { 
  0% {left: 0%;} 
  50% {left: 50%;} 
  100% {left: 100%;} 
}
@keyframes k1 { 
  0% {left: 0%;} 
  50% {left: 50%;} 
  100% {left: 100%;} 
}
@-webkit-keyframes k2 { 
  from {opacity: 0;} 
  to {opacity: 1;} 
}
@-moz-keyframes k2 { 
  from {opacity: 0;} 
  to {opacity: 1;} 
}
@keyframes k2 { 
  from {opacity: 0;} 
  to {opacity: 1;} 
}
.a {animation-name: k1, unknown, k2;}"
`;

exports[`FelaAdapter LTR style sheet handles inline @keyframes 2`] = `"a"`;

exports[`FelaAdapter LTR style sheet handles multiple selectors (comma separated) 1`] = `
".a {cursor: pointer;}
.b:disabled {cursor: default;}
.c[disabled] {cursor: default;}
.d> span {cursor: default;}"
`;

exports[`FelaAdapter LTR style sheet handles multiple selectors (comma separated) 2`] = `"a b c d"`;

exports[`FelaAdapter LTR style sheet handles nested @media 1`] = `
".a {color: red;}
@media (min-width: 300px) { .b {color: blue;} }
@media (min-width: 300px) and (max-width: 1000px) { .c {color: green;} }"
`;

exports[`FelaAdapter LTR style sheet handles nested @media 2`] = `"a b c"`;

exports[`FelaAdapter LTR style sheet handles properties 1`] = `
".a {color: black;}
.b {display: inline;}
.c {margin-right: 10px;}
.d {padding: 0;}"
`;

exports[`FelaAdapter LTR style sheet handles properties 2`] = `"a b c d"`;

exports[`FelaAdapter LTR style sheet handles pseudo selectors 1`] = `
".a {position: fixed;}
.c::before {position: absolute;}
.b:hover {position: static;}"
`;

exports[`FelaAdapter LTR style sheet handles pseudo selectors 2`] = `"a b c"`;

exports[`FelaAdapter LTR style sheet handles raw CSS 1`] = `
".felaadapter-button {display: 'block'; font-size: 16px; text-align: 'left';}
.felaadapter-button:hover {color: 'red';}
.felaadapter-button {vertical-align: 'middle';}
@media (max-width:600px) {.felaadapter-button {display: 'none';}}
@supports (display:-webkit-box) or (display:-webkit-flex) or (display:-ms-flexbox) or (display:flex) {.felaadapter-button {display: 'flex';}}"
`;

exports[`FelaAdapter LTR style sheet handles raw CSS 2`] = `""`;

exports[`FelaAdapter RTL global sheet flushes and purges global styles from the DOM 1`] = `
"body {margin: 0;}
html {height: 100%;}
a {color: red;}
ul {margin: 0;}"
`;

exports[`FelaAdapter RTL global sheet flushes and purges global styles from the DOM 2`] = `""`;

exports[`FelaAdapter RTL global sheet flushes and purges global styles from the DOM 3`] = `""`;

exports[`FelaAdapter RTL style sheet converts unified syntax to native syntax and transforms to a class name 1`] = `
"@-webkit-keyframes k1 { 
  from {opacity: 0;} 
  to {opacity: 1;} 
}
@-moz-keyframes k1 { 
  from {opacity: 0;} 
  to {opacity: 1;} 
}
@keyframes k1 { 
  from {opacity: 0;} 
  to {opacity: 1;} 
}
@font-face {font-family: \\"Roboto\\"; font-style: normal; font-weight: normal; src: url('fonts/Roboto.woff2') format('woff2'),url('fonts/Roboto.ttf') format('truetype');}
.a {margin: 0;}
.b {padding: 6px 12px;}
.c {border: 1px solid #2e6da4;}
.d {border-radius: 4px;}
.e {display: inline-block;}
.f {cursor: pointer;}
.g {font-family: Roboto;}
.h {font-weight: normal;}
.i {line-height: normal;}
.j {white-space: nowrap;}
.k {text-decoration: none;}
.l {text-align: right;}
.m {background-color: #337ab7;}
.n {vertical-align: middle;}
.o {color: rgba(0, 0, 0, 0);}
.p {animation-name: k1;}
.q {animation-duration: .3s;}
.t::before {content: \\"★\\";}
.u::before {display: inline-block;}
.v::before {vertical-align: middle;}
.w::before {margin-left: 5px;}
.r:hover {background-color: #286090;}
.s:hover {border-color: #204d74;}
@media (max-width: 600px) { .x {padding: 4px 8px;} }"
`;

exports[`FelaAdapter RTL style sheet converts unified syntax to native syntax and transforms to a class name 2`] = `"a b c d e f g h i j k l m n o p q r s t u v w x"`;

exports[`FelaAdapter RTL style sheet flushes and purges all styles from the DOM 1`] = `".a {display: block;}"`;

exports[`FelaAdapter RTL style sheet flushes and purges all styles from the DOM 2`] = `"a"`;

exports[`FelaAdapter RTL style sheet flushes and purges all styles from the DOM 3`] = `""`;

exports[`FelaAdapter RTL style sheet handles @fallbacks 1`] = `
".a {background: linear-gradient(...);}
.b {display: flex;}
.c {color: blue;}"
`;

exports[`FelaAdapter RTL style sheet handles @fallbacks 2`] = `"a b c"`;

exports[`FelaAdapter RTL style sheet handles @media 1`] = `
".a {color: red;}
.b {padding-right: 10px;}
@media (min-width: 300px) { .c {color: blue;}
.d {padding-right: 15px;} }
@media (max-width: 1000px) { .e {color: green;}
.f {padding-right: 20px;} }"
`;

exports[`FelaAdapter RTL style sheet handles @media 2`] = `"a b c d e f"`;

exports[`FelaAdapter RTL style sheet handles @supports 1`] = `
".a {display: block;}
@supports not (display: flex) {.c {float: right;}}
@supports (display: flex) {.b {display: flex;}}"
`;

exports[`FelaAdapter RTL style sheet handles @supports 2`] = `"a b c"`;

exports[`FelaAdapter RTL style sheet handles attribute selectors 1`] = `
".a {display: block;}
.b[disabled] {opacity: 0.5;}"
`;

exports[`FelaAdapter RTL style sheet handles attribute selectors 2`] = `"a b"`;

exports[`FelaAdapter RTL style sheet handles descendant selectors 1`] = `
".a {margin: 0;}
.b {padding: 0;}
.c> li {list-style: bullet;}"
`;

exports[`FelaAdapter RTL style sheet handles descendant selectors 2`] = `"a b c"`;

exports[`FelaAdapter RTL style sheet handles inline @font-face 1`] = `
"@font-face {font-family: \\"Roboto\\"; font-style: normal; font-weight: normal; src: url('fonts/Roboto.woff2') format('woff2'),url('fonts/Roboto.ttf') format('truetype');}
@font-face {font-family: \\"Circular\\"; font-style: normal; font-weight: normal; src: url('fonts/Circular.woff2') format('woff2');}
@font-face {font-family: \\"Circular\\"; font-style: italic; font-weight: normal; src: url('fonts/Circular-Italic.woff2') format('woff2');}
@font-face {font-family: \\"Circular\\"; font-style: normal; font-weight: 300; src: url('fonts/Circular-Light.woff2') format('woff2');}
@font-face {font-family: \\"Circular\\"; font-style: normal; font-weight: 700; src: url('fonts/Circular-Bold.woff2') format('woff2');}
.a {font-family: Circular, OtherFont, Roboto;}"
`;

exports[`FelaAdapter RTL style sheet handles inline @font-face 2`] = `"a"`;

exports[`FelaAdapter RTL style sheet handles inline @keyframes 1`] = `
"@-webkit-keyframes k1 { 
  0% {right: 0%;} 
  50% {right: 50%;} 
  100% {right: 100%;} 
}
@-moz-keyframes k1 { 
  0% {right: 0%;} 
  50% {right: 50%;} 
  100% {right: 100%;} 
}
@keyframes k1 { 
  0% {right: 0%;} 
  50% {right: 50%;} 
  100% {right: 100%;} 
}
@-webkit-keyframes k2 { 
  from {opacity: 0;} 
  to {opacity: 1;} 
}
@-moz-keyframes k2 { 
  from {opacity: 0;} 
  to {opacity: 1;} 
}
@keyframes k2 { 
  from {opacity: 0;} 
  to {opacity: 1;} 
}
.a {animation-name: k1, unknown, k2;}"
`;

exports[`FelaAdapter RTL style sheet handles inline @keyframes 2`] = `"a"`;

exports[`FelaAdapter RTL style sheet handles multiple selectors (comma separated) 1`] = `
".a {cursor: pointer;}
.b:disabled {cursor: default;}
.c[disabled] {cursor: default;}
.d> span {cursor: default;}"
`;

exports[`FelaAdapter RTL style sheet handles multiple selectors (comma separated) 2`] = `"a b c d"`;

exports[`FelaAdapter RTL style sheet handles nested @media 1`] = `
".a {color: red;}
@media (min-width: 300px) { .b {color: blue;} }
@media (min-width: 300px) and (max-width: 1000px) { .c {color: green;} }"
`;

exports[`FelaAdapter RTL style sheet handles nested @media 2`] = `"a b c"`;

exports[`FelaAdapter RTL style sheet handles properties 1`] = `
".a {color: black;}
.b {display: inline;}
.c {margin-left: 10px;}
.d {padding: 0;}"
`;

exports[`FelaAdapter RTL style sheet handles properties 2`] = `"a b c d"`;

exports[`FelaAdapter RTL style sheet handles pseudo selectors 1`] = `
".a {position: fixed;}
.c::before {position: absolute;}
.b:hover {position: static;}"
`;

exports[`FelaAdapter RTL style sheet handles pseudo selectors 2`] = `"a b c"`;

exports[`FelaAdapter RTL style sheet handles raw CSS 1`] = `
".felaadapter-button {display: 'block'; font-size: 16px; text-align: 'right';}
.felaadapter-button:hover {color: 'red';}
.felaadapter-button {vertical-align: 'middle';}
@media (max-width:600px) {.felaadapter-button {display: 'none';}}
@supports (display:-webkit-box) or (display:-webkit-flex) or (display:-ms-flexbox) or (display:flex) {.felaadapter-button {display: 'flex';}}"
`;

exports[`FelaAdapter RTL style sheet handles raw CSS 2`] = `""`;
